src/extension.ts

```diff
@@ -124,13 +124,20 @@ export async function activate(context: vscode.ExtensionContext) {
 						`${CODEMOD_METADATA_SCHEME}:${name}.md`,
 					);
 
-					const metadataExist = textContentProvider.hasMetadata(uri);
-
-					if (!metadataExist) {
+					if (!textContentProvider.hasMetadata(uri)) {
 						return;
 					}
 
-					vscode.commands.executeCommand('markdown.showPreview', uri);
+					const metadataContent =
+						textContentProvider.provideTextDocumentContent(uri);
+
+					store.dispatch(
+						actions.setPanelView({
+							kind: 'CODEMOD',
+							title: `${name}.md`,
```

Comment:
we can provide the codemod name as the title, we definitely don't want the `.md` to be visible (we needed it only for the native markdown panel to work)